lib/bing_translator.rb

```diff
@@ -42,7 +42,22 @@ def translate(text, params = {})
       'contentType' => params[:content_type] || 'text/plain'
     }
 
-    result(:translate, params).body[:translate_response][:translate_result]
+    result(:translate, params)
+  end
+
+  def translate_array(texts, params = {})
+    raise "Must provide :to." if params[:to].nil?
+
+    # Important notice: param order makes sense in SOAP. Do not reorder or delete!
+    params = {
+      'texts'       => { 'arr:string' => texts },
+      'from'        => params[:from].to_s,
+      'to'          => params[:to].to_s,
+      'category'    => 'general',
+      'contentType' => params[:content_type] || 'text/plain'
+    }
+
+    array_wrap(result(:translate_array, params)[:translate_array_response]).map{|r| r[:translated_text]}
```

Comment:
A single entry if the texts array only contains 1 element. That's why I had to use `array_wrap` which is a copy of Array#wraps defined in Rails (see http://apidock.com/rails/Array/wrap/class).